### ISSUE TITLE
[1.12.x] Duplicated "VolumeMounts" Keys in 1-gloo-deployment.yaml

### ISSUE THREAD

**Issue by vmarlier** [solo-io/gloo#6977]
### Gloo Edge Version

1.12.x (latest stable)

### Kubernetes Version

1.22.x

### Describe the bug

Hello,
During our attempt to update Gloo to 1.12.6 with our Fluxv2 stack, we noticed an issue outputted by the `Helm-Controller`.
Here is the error:
```
Helm upgrade failed: error while running post render on files: map[string]interface {}(nil): yaml: unmarshal errors:
        line 101: mapping key "volumeMounts" already defined at line 75
```
Basically Helm is complaining about a duplicated `volumeMounts` key.

The issue is only firing when we want to install through the flux stack since they added a check for duplated keys. When using the helm cli, it seems helm do not complain but take randomly one of the VolumeMounts key definition. So it is still an issue.

When taking a look at the file for 1.12.x I can see a duplication [here](https://github.com/solo-io/gloo/blob/e46bcf6ccf791ff65b0b73f67dec8e2f39bbd4e2/install/helm/gloo/templates/1-gloo-deployment.yaml#L195) and [here](https://github.com/solo-io/gloo/blob/e46bcf6ccf791ff65b0b73f67dec8e2f39bbd4e2/install/helm/gloo/templates/1-gloo-deployment.yaml#L253).

It is possible to fix this please ?

### Steps to reproduce the bug

1. Install Gloo via Fluxv2 stack with a version prior to 1.12.x
2. Update to 1.12.x
3. Should see an issue with the fluxv2 helm-controller.

### Expected Behavior

Being able to install it through fluxv2 without any blocker.

**Comment by kevin-shelaga** [comment]
looks to be an issue when `gateway.validation.enabled` and `gateway.enabled` are true

**Comment by sam-heilbron** [comment]
This has been resolved in affected versions (1.12, 1.3) of Gloo.
